runtime/service.go

```diff
@@ -1149,54 +1135,70 @@ func (s *service) Connect(requestCtx context.Context, req *taskAPI.ConnectReques
 	return nil, errdefs.ErrNotImplemented
 }
 
-// Shutdown will attempt a graceful shutdown of the shim+VM. The shutdown procedure will only actually take
-// place if "Now" was set to true OR the VM started successfully, all tasks have been deleted and we were
-// told to shutdown when all tasks were deleted. Otherwise the call is just ignored.
+// Shutdown will shutdown of the VMM. Unlike StopVM, this method is only exposed to containerd itself.
+//
+// The shutdown procedure will only actually take place if "Now" was set to true OR
+// the VM started successfully, all tasks have been deleted and we were told to shutdown when all tasks were deleted.
+// Otherwise the call is just ignored.
 //
-// Shutdown can be called from a few code paths:
-// * If StopVM is called by the user (in which case "Now" is set to true)
-// * After any task is deleted via containerd's API (containerd calls on behalf of the user)
-// * After any task Create call returns an error (containerd calls on behalf of the user)
-// Shutdown is not directly exposed to containerd clients.
+// containerd calls this API on behalf of the user in the following cases:
+// * After any task is deleted via containerd's API
+// * After any task Create call returns an error
 func (s *service) Shutdown(requestCtx context.Context, req *taskAPI.ShutdownRequest) (*ptypes.Empty, error) {
 	defer logPanicAndDie(log.G(requestCtx))
-	s.logger.WithFields(logrus.Fields{"task_id": req.ID, "now": req.Now}).Debug("shutdown")
+	s.logger.WithFields(logrus.Fields{"task_id": req.ID, "now": req.Now}).Debug("Shutdown")
 
 	shouldShutdown := req.Now || s.exitAfterAllTasksDeleted && s.taskManager.ShutdownIfEmpty()
 	if !shouldShutdown {
 		return &ptypes.Empty{}, nil
 	}
 
-	// cancel the shim context no matter what, which will result in the VM getting a SIGKILL (if not already
-	// dead from graceful shutdown) and the shim process itself to begin exiting
-	defer s.shimCancel()
-
 	s.logger.Info("stopping the VM")
```

Comment:
nit: this log could probably be put in `shutdown` now

Reply:
Will move.

Reply:
Moved